projects/packages/image-cdn/src/class-image-cdn.php

```diff
@@ -336,297 +367,342 @@ public static function parse_dimensions_from_filename( $src ) {
-						 *
-						 * @module photon
-						 *
-						 * @since 2.0.3
-						 *
-						 * @param bool false Was the image uploaded to the local site. Default to false.
-						 * @param array $args {
-						 *   Array of image details.
-						 *
-						 *   @type $src Image URL.
-						 *   @type tag Image tag (Image HTML output).
-						 *   @type $images Array of information about the image.
-						 *   @type $index Image index.
-						 * }
-						 */
-						apply_filters( 'jetpack_photon_image_is_local', false, compact( 'src', 'tag', 'images', 'index' ) )
-					) {
-						$attachment_id = (int) array_pop( $attachment_id );
-
-						if ( $attachment_id ) {
-							$attachment = get_post( $attachment_id );
-
-							// Basic check on returned post object.
-							if ( is_object( $attachment ) && ! is_wp_error( $attachment ) && 'attachment' === $attachment->post_type ) {
-								$src_per_wp = wp_get_attachment_image_src( $attachment_id, isset( $size ) ? $size : 'full' );
-
-								if ( self::validate_image_url( $src_per_wp[0] ) ) {
-									$src          = $src_per_wp[0];
-									$fullsize_url = true;
-
-									// Prevent image distortion if a detected dimension exceeds the image's natural dimensions.
-									if ( ( false !== $width && $width > $src_per_wp[1] ) || ( false !== $height && $height > $src_per_wp[2] ) ) {
-										$width  = false === $width ? false : min( $width, $src_per_wp[1] );
-										$height = false === $height ? false : min( $height, $src_per_wp[2] );
-									}
-
-									// If no width and height are found, max out at source image's natural dimensions.
-									// Otherwise, respect registered image sizes' cropping setting.
-									if ( false === $width && false === $height ) {
-										$width     = $src_per_wp[1];
-										$height    = $src_per_wp[2];
-										$transform = 'fit';
-									} elseif ( isset( $size ) && array_key_exists( $size, $image_sizes ) && isset( $image_sizes[ $size ]['crop'] ) ) {
-										$transform = (bool) $image_sizes[ $size ]['crop'] ? 'resize' : 'fit';
-									}
-								}
-							} else {
-								unset( $attachment_id );
-								unset( $attachment );
-							}
-						}
-					}
+			// Flag if we need to munge a fullsize URL.
+			$fullsize_url = false;
 
-					// If image tag lacks width and height arguments, try to determine from strings WP appends to resized image filenames.
-					if ( false === $width && false === $height ) {
-						list( $width, $height ) = self::parse_dimensions_from_filename( $src );
-					}
+			// Identify image source.
+			$src_orig = $p->get_attribute( 'src' );
+			$src      = $src_orig;
 
-					$width_orig     = $width;
-					$height_orig    = $height;
-					$transform_orig = $transform;
+			/**
+			 * Allow specific images to be skipped by Photon.
+			 *
+			 * @TODO: Does this need to pass the full HTML of the image tag?
```

Comment:
I looked at usage of this filter in other plugins in the WordPress.org plugin directory. While most plugins do not use this third parameter, there are a few that do, sometimes to look for a specific class name in the HTML. So I think we should keep this available if possible.